ApeWorX/silverback
# refactor: allow multiple handlers

### What I did

Refactor to dynamic registration of tasks, saving tasks (and their handlers) in a separate collection that can be accessed via an enum type to specify their class

Relates to https://github.com/ApeWorX/silverback/issues/57, as it migrates to a dynamic registration method via TaskIQ

Also should allow https://github.com/ApeWorX/silverback/pull/55#issuecomment-1977168440 to use dynamic registration to resolve the bug with multiple handlers with different filters not fully registering

### How I did it

New TaskIQ dynamic broker registration API

### How to verify it

Everything should work as it did before, don't think this is breaking but wanted to call attention to commits to make sure they were reviewed fully for breaking behaviors

### Checklist

~~- [ ] New test cases have been added and are passing~~
~~- [ ] Documentation has been updated~~
